### ISSUE TITLE
Decreased visibility for sensitive notifications

### ISSUE THREAD

**Issue by beverloo** [whatwg/notifications#45]
Some mobile platforms show notifications on their lock-screen (prior to the user _logging in_), which is great for a quick peek about what's going on, but not so great for notifications that contain rather sensitive content. In these cases, an implementation could decide to display a more generic/anonymized version instead.

How about adding a `sensitive` (boolean) option, defaulting to false?

@jakearchibald

**Comment by annevk** [comment]
What do you mean by logging in?

**Comment by beverloo** [comment]
Entering one's security code to gain access to the device.

**Comment by annevk** [comment]
Ah, makes sense. Though typically this is up to the user to decide, no? Or do iOS and Android also expose this?

**Comment by mvano** [comment]
On Android you can set [visibility](http://developer.android.com/reference/android/app/Notification.html#visibility) to public (all content visible), secret (no notification visible at all), or private (redacted visibility) which is the default. It also provides an optional way to specify what is visible for private, with a fallback to some reasonable default representation. I think we wouldn't want to expose all of that in this spec, but a simple boolean to trigger "secret" seems reasonable.

The user can set device-wide preferences as well.

**Comment by annevk** [comment]
I guess that makes sense. "sensitive" is a little hard to type. "secret" works. @sicking?

**Comment by sicking** [comment]
I don't work on these things any more so I can't really provide implementation feedback.

However the feature sounds sensible to me. Though sounds like we should have more than just a boolean so that you can control more than just "show vs. hide" on the lockscreen?

**Comment by mvano** [comment]
Yes an enum would capture the Android feature set better. I'm not sure what, if anything, other platforms expose for this. It would be nice to have at least one other platform to compare and contrast with.
